FV3/makefile

```diff
@@ -240,11 +227,42 @@ cleanall: clean
 	$(RM) -f conf/configure.fv3
 	$(RM) -rf $(SERIALBOX_OUTDIR)
 
-MKDEPENDS = ./mkDepends.pl
-include conf/make.rules
 
 # do not include 'depend' file if the target contains string 'clean'
 ifneq (clean,$(findstring clean,$(MAKECMDGOALS)))
     -include depend
 endif
 
+# Wrapper related build rules
+TEMPLATES_RELATIVE =  dynamics_data.F90 flagstruct_data.F90 physics_data.F90 _wrapper.pyx
+TEMPLATES_IN = $(addprefix wrapper/templates/,$(TEMPLATES_RELATIVE))
+TEMPLATES_OUT = $(addprefix wrapper/lib/,$(TEMPLATES_RELATIVE))
+WRAPPER_OBJECTS=$(TEMPLATES_OUT:.F90=.o) wrapper/lib/coupler_lib.o
+PROPERTIES_FILES = wrapper/fv3gfs/wrapper/dynamics_properties.json \
+	wrapper/fv3gfs/wrapper/flagstruct_properties.json \
+	wrapper/fv3gfs/wrapper/physics_properties.json
+SETUP_PY_LIBS="$(TOP_LEVEL_OBJS_ABSOLUTE) -L$(PWD) $(FV3_LIB_FLAGS) $(NCEPLIBS) $(LDFLAGS) -lfmpich -lmpich -lmvec -lm"
+
+wrapper_clean:
+	cd wrapper/lib && rm -f *.o *.mod *.lst *.c depend $(TEMPLATES_RELATIVE)
+
+wrapper_build:$(WRAPPER_OBJECTS)
+	echo $(WRAPPER_OBJECTS)
+	cd wrapper && SETUP_PY_LIBS=$(SETUP_PY_LIBS) python3 setup.py build_ext --inplace
+	cd wrapper && SETUP_PY_LIBS=$(SETUP_PY_LIBS) python3 setup.py bdist_wheel
+
+$(TEMPLATES_OUT): $(TEMPLATES_IN) $(PROPERTIES_FILES)
+	# fill_templates.py needs to be run in the same directory as the outputs
+	cd wrapper/lib && python3 ../fill_templates.py $(TEMPLATES_RELATIVE)
+
+# Specify dependency structure of the wrapper object files. Would be easier to
+# include in the DEPEND_FILES macro above so that mkDepends.pl could generate
+# the dependency graph. This would however, setup a chicken or egg problem where
+# wrapper/fill_templates.py would need to be run before generating running
+# mkDepends.pl. And we don't want to do that for non-wrapper builds.
+wrapper/lib/physics_data.o: wrapper/lib/physics_data.F90 wrapper/lib/dynamics_data.o  atmos_model.o
+wrapper/lib/coupler_lib.o: wrapper/lib/coupler_lib.F90 wrapper/lib/dynamics_data.o atmos_model.o
+wrapper/lib/flagstruct_data.o: wrapper/lib/flagstruct_data.F90 atmos_model.o
+wrapper/lib/dynamics_data.o: wrapper/lib/dynamics_data.F90 atmos_model.o
+
+include conf/make.rules
```

Comment:
nit: does it make more sense to put this wrapper build logic in the `wrapper/Makefile`?